Imports.md

```diff
@@ -0,0 +1,407 @@
+# ![wgsl-import-spec](branding.svg)
+
+# Summary
+
+We propose adding an importing mechanism to the WGSL shading language as an extension.
+
+# Motivation
+
+When writing bigger WGSL shaders, one tends to **split the code into reusable pieces**. Examples are re-using a set of lighting calculation functions and sharing a struct between two compute shaders and.
+
+However, current WGSL tooling is not built for this. The official language purposefully does not include this feature, nor does it provide adjacent features like namespaces, and there is no standardized extension yet.
+
+We also should account for **importing shader from libraries**. Ideally, users could upload WGSL shaders to existing package managers, which other users could then consume.
+
+Finally, we want **multiple tools** which can compile WGSL-with-imports down to raw WGSL. Using WGSL-with-imports both in Rust projects and in web projects should be possible.
+
+## Guide-level explanation
+
+The `import` statement extension is designed to appear somewhat familiar to TypeScript syntax,
+but with a Rust like recursive grammar to make importing
+multiple importable items less verbose.
+The syntax is also heavily inspired by [Gleam](https://gleam.run/).
+
+By placing an import at the very top of a file, one can either import an entire module, or only specific importable items, such as functions, structs or types.
+
+```
+// Importing a single item using a relative path
+import ./lighting/pbr;
+
+// Importing multiple items
+import my/geom/sphere/{ draw, default_radius as foobar };
+
+// Imports a whole module. Use it with `bevy_ui.name`
+import bevy_ui/*;
+```
+
+These can then be used anywhere in the source code.
+
+```
+fn main() {
+    bevy_ui.quad(vec2f(0.0, 1.0), 1.0);
+    let a = draw(3);
+}
+```
+
+Both `bevy_ui` and `my` are packages in the current project. Language servers and related tools can look in a `wgsl.toml` file to find the location of the packages. This lets libraries be published to package managers, and users can import them with a simple syntax.
+
+The first part is the file path, which is assumed to have a either `.wgsl` file extension,
+or the extension we use for extended wgsl syntax (possibly `.wesl`).
+
+Recursive import definitions are also supported, which leads to short and succinct import statements.
+
+```
+import bevy_pbr/{
+  forward_io/VertexOutput,
+  pbr_types/{PbrInput, pbr_input_new},
+  pbr_bindings/*
+};
+
+fn main() {
+
+}
+```
+
+# Reference-level explanation
+
+Imports must appear as the first items in a WGSL file. They import "importable items" (see [GLOSSARY.md](./GLOSSARY.md)).
+
+An import statement is parsed as follows, with spaces and comments allowed between tokens:
+
+```
+main:
+| 'import' import_path ';'
+
+import_path:
+| ('.' | '..') '/' import_path
+| ident '/' (import_path | import_collection | item_import | star_import)
```

Comment:
I think we could restrict the dots by writing

```
main:
| 'import' import_relative? import_path ';'

import_relative:
| ('.' | '..') '/' ('..' '/')*

import_path:
| ident '/' (import_path | import_collection | item_import | star_import)

```

Reply:
you're a grammar whiz. adding